api/envoy/type/matcher/v3/string.proto

```diff
@@ -61,6 +62,11 @@ message StringMatcher {
     //
     // * ``abc`` matches the value ``xyz.abc.def``
     string contains = 7 [(validate.rules).string = {min_len: 1}];
+
+    // Use an extension as the matcher type.
+    // [#extension-category: envoy.matching.string]
+    // [#comment: This uses the xds type instead of the envoy type to avoid a circular dependency in the build system.]
+    xds.core.v3.TypedExtensionConfig extension = 8;
```

Comment:
Because StringMatcher is used in many places where the new matchers are not supported, and adding this here gets support in all those places.

I'm not very familiar with the new matcher API, but I looked at it briefly and it seems to always have actions associated with a match; I'm not exactly sure how to apply that in some cases where StringMatcher is used where it just needs a boolean response.

One specific case I want to use this extension matcher is in SAN validation in the tls transport socket.

Reply:
Yeah, I think I'm fine conceptually here with this being an extension type for the reasons Greg gives.